### PR TITLE
Limit diff size to prevent out of control memory use.

### PR DESCRIPTION
**Pull request by bill-rich**

### CODE REVIEW

File: pkg/gitparse/gitparse.go [hunk 1]
```diff
@@ -20,6 +20,9 @@ import (
 // DateFormat is the standard date format for git.
 const DateFormat = "Mon Jan 02 15:04:05 2006 -0700"
 
+// MaxDiffSize is the maximum size for a diff. Larger diffs will be cut off.
```
Comment (ahrav) [inline pkg/gitparse/gitparse.go:23]: nit: We can group the two consts together?

const (
    // DateFormat is the standard date format for git.
    DateFormat = "Mon Jan 02 15:04:05 2006 -0700"

    // MaxDiffSize is the maximum size for a diff. Larger diffs will be cut off.
    MaxDiffSize = 1 * 1024 * 1024 * 1024 // 1GB
)

File: pkg/gitparse/gitparse_test.go [hunk 2]
```diff
@@ -181,6 +181,30 @@ func TestMultiCommitContextDiff(t *testing.T) {
 	}
 }
 
+func TestMaxDiffSize(t *testing.T) {
+	bigBytes := bytes.Buffer{}
```
Comment (ahrav) [inline pkg/gitparse/gitparse_test.go:185]: bigBytes !! 😆

**Review by ahrav — APPROVED** [review]
Nicee.

One question, is there a plan to handle big diffs, or are we just ignoring them?

**Comment by bill-rich** [comment]
> Nicee.
> 
> One question, is there a plan to handle big diffs, or are we just ignoring them?

For now, just ignoring them. Long term, I think something like a listing of files/repos/orgs etc that were not 100% scanned.

**Pull request merged**